pkg/controller/utils/gcp/utils.go

```diff
@@ -151,18 +151,34 @@ func CheckPermissionsAgainstPermissionList(gcpClient ccgcp.Client, permList []st
 
 	filteredPermList := filterOutPermissions(permList, invalidPermsForProjectScopedPermissionsTesting)
 
-	permRequest := &cloudresourcemanager.TestIamPermissionsRequest{
-		Permissions: filteredPermList,
-	}
-	permResponse, err := gcpClient.TestIamPermissions(projectName, permRequest)
-	if err != nil {
-		return false, fmt.Errorf("error testing permissions: %v", err)
+	// Split list to only check 100 permissions at a time
+	responses := []*cloudresourcemanager.TestIamPermissionsResponse{}
+	done := false
+	index := 0
+	for !done {
+		upperLimit := len(filteredPermList)
+		if index+100 < upperLimit {
+			upperLimit = index + 100
+		} else {
+			done = true
+		}
+		permRequest := &cloudresourcemanager.TestIamPermissionsRequest{
+			Permissions: filteredPermList[index:upperLimit],
+		}
+		permResponse, err := gcpClient.TestIamPermissions(projectName, permRequest)
+		if err != nil {
+			return false, fmt.Errorf("error testing permissions: %v", err)
+		}
+		responses = append(responses, permResponse)
+		index += 100
 	}
```

Comment:
Just another possible approach for consideration since I found the problem interesting:

```go
var chunks [][]string
chunkSize := 100
permLen := len(filteredPermList)
for i := 0; i < permLen; i += chunkSize {
    end := i + chunkSize
    if end > permLen {
        end = permLen
    }
    chunks = append(chunks, filteredPermList[i:end])
}

for chunk := range chunks {
  // do request
}
```

Reply:
Thx! I used it for the test as an alternative way to come up with the chunks